proof/refine/ARM/CNodeInv_R.thy

```diff
@@ -7026,20 +7092,26 @@ next
            apply (clarsimp simp: capRemovable_def cte_wp_at_ctes_of)
           apply (clarsimp dest!: isCapDs simp: cte_wp_at_ctes_of)
           apply (case_tac "cteCap rv'",
-                 auto simp add: isCap_simps is_cap_simps final_matters'_def)[1]
-         apply (wp isFinalCapability_inv static_imp_wp
-                 | simp add: is_final_cap_def conj_comms cte_wp_at_eq_simp)+
-       apply (rule isFinal[where x="cte_map slot"])
-      apply (wp get_cap_wp| simp add: conj_comms)+
-      apply (wp getCTE_wp')
+                 auto simp add: isCap_simps is_cap_simps final_matters'_def pred_tcb_at'_def
```

Comment:
Is the auto necessary here?

Reply:
Weirdly, yes (you can see its predecessor at line 7029 above). IIRC there's a schematic involved that `force` and friends don't want to instantiate, but `auto` is happy to.

Reply:
Blerg